### PR TITLE
feat: adds stats to header of how-to and research

### LINKED ISSUE #2734: [feature request] Add stats to header How-to and Research
### Description
Add space to include more statistics in the header of the research and how-to's

Stats to include:

**Research**
- Viewcount (currently live in beta)
- Users Following (stat is not shown yet)
- Useful (stat is live on research overview)
- Comments (stat is live on research overview)
- Updates (stat is live on research overview)

**How-to**
- Viewcount (currently live in beta)
- Useful (stat is live on how-to overview)
- Comments (stat is not shown yet)
- Steps (stat is not shown yet)

**Notes:**
- This allows in the future to make "useful" and "following" clickable so you can see which users enagegd
- There will be another issue to follow up **mobile** (#2735)
- All icons are already available on live. 

### Page related

Will be used in :
[Research](https://dev.onearmy.world/research/more-draft-checking) 
[How-to](https://dev.onearmy.world/how-to/test-title)

### Mockup
afbeelding

afbeelding

### PR DESCRIPTION
**Pull request by CheRayLiu**
PR Checklist

- [x] - Commit [messages are descriptive](https://github.com/ONEARMY/community-platform/blob/master/CONTRIBUTING.md#--commit-style-guide), it will be used in our [Release Notes](https://github.com/ONEARMY/community-platform/releases/)

PR Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Developer experience (improves developer workflows for contributing to the project)

## Description

Add statistics to header of research and how-to by:
- Creating components for `ResearchStatistics` and `HowToStatistics`
- Using stats component in Research and HowTo header
- Adding statistics methods in `/stores`
- Added cypress tests for both components

## Git Issues

Closes #2734

## Screenshots/Videos
Screenshot 2023-09-03 at 7 19 29 PM
Screenshot 2023-09-03 at 7 19 17 PM

## What happens next?

Thanks for the contribution! We try to make sure all PRs are reviewed ahead of a monthly dev call (first Monday of the month, open to all!).

If the PR is working as intended it'll be merged and included in the next platform release, if not changes will be requested and re-reviewed once updated.

If you need more immediate feedback you can try reaching out on Discord in the [Community Platform `development` channel](https://discord.com/channels/586676777334865928/938781727017558018).

### CODE REVIEW

**Comment by davehakkens** [comment]
Hey @CheRayLiu are you able to add the tests? And could you let us know when its out of draft and ready for review. Would be great to see this merged!

**Comment by CheRayLiu** [comment]
Hey @davehakkens it's ready for review! However, I haven't had time to add the tests. Will try to get to it by the end of the week.

**Comment by davehakkens** [comment]
nice! @thisislawatts 👆

**Comment by CheRayLiu** [comment]
Just added tests!

File: packages/components/src/Icon/Icon.tsx [hunk 1]
```diff
@@ -155,7 +156,7 @@ export const Icon = (props: Props) => {
   return (
     <IconWrapper
       {...props}
-      sx={{ color: props.color ?? 'inherit' }}
+      sx={{ color: props.color ?? 'inherit', opacity: props.opacity }}
```
Comment (thisislawatts) [inline packages/components/src/Icon/Icon.tsx:159]: **question(blocking):** As props.opacity is optional should we provide a fallback or default value for `opacity`?

File: packages/components/src/HowToStatistics/HowToStatistics.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  viewCount: number
+  usefulCount: number
+  commentCount: number
+  stepCount: number
+}
+
+const ICON_OPACITY = '0.5'
+const FONT_SIZE = '1'
+
+export const HowToStatistics = (props: IProps) => (
```
Comment (thisislawatts) [inline packages/components/src/HowToStatistics/HowToStatistics.tsx:14]: **issue(blocking):** Generally speaking we should aim for our components to be reusable across multiple places in the Platform. It looks like the 2 new components you have introduced here share very similar traits. So in theory you could refactor them to be a single component. 

Please take a look at a sketch I have added in https://github.com/ONEARMY/community-platform/pull/2737/commits/9b7110226b74d5d33a203fa1071bb2adf5dbdfe0 which introduces a generic `ContentStatistic` component. This does push the plural label formatting out to the controllers. What do you think of this approach?

File: packages/cypress/src/integration/research/read.spec.ts [hunk 3]
```diff
@@ -55,6 +55,11 @@ describe('[Research]', () => {
         cy.step(`ViewsCounter should not be visible`)
         cy.get('[data-cy="ViewsCounter"]').should('not.exist')
       })
+
+      it('[ResearchStatistics only visible for beta-testers]', () => {
+        cy.step(`ResearchStatistics should not be visible`)
+        cy.get('[data-cy="ResearchStatistics"]').should('not.exist')
+      })
```
Comment (thisislawatts) [inline packages/cypress/src/integration/research/read.spec.ts:62]: **praise:** Great to add these e2e checks around beta features, thank you!

**Review by thisislawatts — CHANGES_REQUESTED** [review]

File: packages/components/src/Icon/Icon.tsx (hunk 1 above)
Comment (CheRayLiu) [reply packages/components/src/Icon/Icon.tsx:159]: Good call! I'll update it to default to '1' as that's the [default value of opacity](https://developer.mozilla.org/en-US/docs/Web/CSS/opacity)

File: packages/components/src/HowToStatistics/HowToStatistics.tsx (hunk 2 above)
Comment (CheRayLiu) [reply packages/components/src/HowToStatistics/HowToStatistics.tsx:14]: Yeah, I like that approach better to use `ContentStatistic`, thanks for the suggestion! 

I can create a reusable util method to handle the plural label logic there

Comment (CheRayLiu) [reply packages/components/src/HowToStatistics/HowToStatistics.tsx]: Updated!

**Comment by davehakkens** [comment]
Is this one be ready to merge @CheRayLiu @thisislawatts ?
Would help to build https://github.com/ONEARMY/community-platform/pull/2802

**Comment by CheRayLiu** [comment]
Yep ready for merge from my perspective! Just waiting for additional reviews or approval

**Comment by thisislawatts** [comment]
This is looking great @CheRayLiu, would you be able to resolve the merge conflicts? ThenI think we are in a place to merge this change ✨

**Review by thisislawatts — APPROVED** [review]

**Comment by CheRayLiu** [comment]
Hey all, I fixed the merge conflicts and I think it's ready for merge!

However, it is failing due to an old commit message failing lint and I cannot find the commit to amend it in the commit log even using the sha shown on GitHub UI. Would it possible to override this lint commit rule?

**Comment by AlfonsoGhislieri** [comment]
Hi @CheRayLiu, the issue is with [3e5603b](https://github.com/ONEARMY/community-platform/pull/2737/commits/3e5603bb0e46036d65ab7c0d00a3900f6450009e):

`✖   subject must not be sentence-case, start-case, pascal-case, upper-case [subject-case]
`
You could rebase interactively to change the commit and force push, as is explained in the github documenation:
https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/changing-a-commit-message

**Comment by CheRayLiu** [comment]
Hey @AlfonsoGhislieri Yep I am aware of the issue and I've tried to do that previously already for a few hours looking various ways. I cannot find the specific commit (3e5603bb0e46036d65ab7c0d00a3900f6450009e) to amend to changes. 

I will try again

**Comment by CheRayLiu** [comment]
Hey @AlfonsoGhislieri Yes I am aware of the issue and I've tried to rebase interactively. I think the issue is that since it's the first commit of the branch, it doesn't show up when I do `git rebase -i ...` so I cannot reword the git message. 

I'll keep trying to search for a way resolve this.

**Comment by thisislawatts** [comment]
Sorry for the extra hassle here @CheRayLiu, it is frustrating when git gets in the way like this.

**Comment by thisislawatts** [comment]
Thank you so much @CheRayLiu for all the time and effort 💚, working with the changes you introduced here I created https://github.com/ONEARMY/community-platform/pull/2911. I've got you as a co-author for the changes.

**Comment by CheRayLiu** [comment]
No problem, apologies for not being able to push it through the finish line. Appreciate the help here! 🙏

**Pull request closed without merging**